### ISSUE TITLE
Expressions: New simple functions

### ISSUE THREAD

**Issue by olemartinorg** [Altinn/app-frontend-react#1176]
Extracted from #646.

We should implement some new simple functions in the expression language. All of these implementations should be fairly easy, but if any of these presents problems during the implementation phase, new issues should be created for them specifically.

List of functions we need:
- [x] `round`
  - Rounds a number to an integer, or optionally a decimal with a configurable amount of decimal points
- [x] `text`
  - Should look up a value from a text resource, or from a language key. Additional arguments should be considered parameters to the text resource, if any are provided. Returns `string` or `null`
- [x] `strLength` / `stringLength` / `numCharacters`
  - Function taking a sting as an argument and returns the number for characters in the string (note: the number of characters is not the same as the number for bytes)
- [x] `language`
  - Function with no parameters, returns the currently selected language (i.e. `nb`, `en`, etc).
- [x] `contains`
  - String-checking function to check if the second parameter (a `string`) is inside the first parameter (also a `string`)
- [x] `notContains`
  - Synonym for `["not", ["contains", ...]]`
- [x] `endsWith`
  - String-checking function to check if the first parameter ends with the string in the second parameter
- [x] `startsWith`
  - Same as the above, but for the beginning of the text
- [x] `commaContains`
  - Special function for checking if a value is included in a comma-separated list of strings.
  - Can also be represented as a shortcut to `["or", ["equals", string, input], ["startsWith", string, ["concat", input, ","]], ["endsWith", string, ["concat", ",", input]], ["contains", string, ["concat", ",", input, ","]]]`
- [ ] `displayValue` **Created a separate [issue](https://github.com/Altinn/app-frontend-react/issues/1208)**
  - Looks up a component nearby, and presents its display value (which should work for number-formatted Input fields). See comment here: https://github.com/Altinn/app-frontend-react/issues/1105#issuecomment-1536240868
- [x] `upperCase`
  - Takes a string, turns it into UPPERCASE
- [x] `lowerCase`
  - Takes a string, turns it into lowercase

List of functions that require [list support](https://github.com/Altinn/app-frontend-react/issues/1179) in the expression language:
- [ ] `sum`
  - Should take any amount of numbers as input, and also lists of numbers. All these numbers should be summed, and the resulting number should be returned. `null` and non-numeric values should be considered to be `0` (i.e. no effect on the sum).
- [ ] `average`
  - Same as above, but should produce the average of all the numbers summed. `null` and non-numeric values should be considered entries when calculating the average, as we should make a filtering function for those cases where invalid values should not be considered instead.
- [ ] `in`
  - Shortcut function similar to `equals`, but compares the first argument against multiple others to check if the first argument matches either one of them. Should support lists of items as parameters, and should then check inside the lists as well.
- [ ] `notIn`
  - Synonym for `["not", ["in", ....]]`

**Comment by framitdavid** [comment]
I decided not to implement displayValue within this issue, due to https://github.com/Altinn/app-frontend-react/issues/1208,

**Comment by bjosttveit** [comment]
Is this the expected behavior? That startsWith, endsWith, and contains always return true for an empty string

![Image](https://github.com/Altinn/app-frontend-react/assets/47412359/013ddbd6-0f31-4bc8-a73a-3b24f730499a)

**Comment by framitdavid** [comment]
> Is this the expected behavior? That startsWith, endsWith, and contains always return true for an empty string
> 
> ![Image](https://github.com/Altinn/app-frontend-react/assets/47412359/013ddbd6-0f31-4bc8-a73a-3b24f730499a)
> 
> Everything seems to work, tested in dev-tools.
> 

Hmm, that looks wired.. I think we should fix this. If I remember correctly this functions is implemented using the JavaScript startsWith etc.

**Comment by olemartinorg** [comment]
@bjosttveit I think it's expected behaviour, because all strings both starts and ends with nothing. It does not make sense to do that, but I still think it should work. If you do a `["startsWith", "myString", ["dataModel", "something"]]` and that lookup in the data model returns an empty string, you'd get `true` back - but you could guard against that by checking for an empty string as well. However, if the data model path does not exist, a `null` will be returned, and after #1217 that should always return `false`.

So, TL;DR: All strings starts/ends with an empty string, but no string starts/ends with a `null`. I think that's the appropriate compromise, as long as we document it.
